cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -390,6 +417,12 @@ struct UnboxScalar<Type, enable_if_has_string_view<Type>> {
   }
 };
 
+template <typename Type>
+struct UnboxScalar<Type, enable_if_list_type<Type>> {
```

Comment:
So fixed_size_list is also declared as scalar, but not being registered in compute? ( It's ok to me, just to make sure this)

Reply:
I can look at adding that to the compute registry